### PR TITLE
Add Aspect Ratio component

### PR DESCRIPTION
**Pull request by alimpens**

### CODE REVIEW

**Review by VincentSmedinga — CHANGES_REQUESTED** [review]

File: packages/css/src/aspect-ratio/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Aspect ratio
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/aspect-ratio/README.md]: Good idea, I'll change it for all docs

File: packages/css/src/aspect-ratio/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Gebruik Aspect Ratio (in het Nederlands: beeldverhouding) om de breedte van een element te koppelen aan de hoogte. Hoe smaller het element, hoe minder hoog.
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/aspect-ratio/README.md:3]: Maybe something like this?
```suggestion
Aspect Ratio (in het Nederlands: beeldverhouding) koppelt de hoogte van een element aan de breedte.
```

File: packages/css/src/aspect-ratio/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+Dit component staat een vaste set verhoudingen (breedte:hoogte) toe, zodat het element dat het bevat een consistent formaat heeft:
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/aspect-ratio/README.md:5]: Hm, the 'staande en liggende modus' part doesn't really make this part clearer, in my opinion. Maybe something like:

```suggestion
Om de vormgeving consistent te houden gebruiken we een vaste set verhoudingen (breedte:hoogte):
```

File: packages/css/src/aspect-ratio/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+- `extra-tall`: 9:16
+- `tall`: 4:5
+- `square`: 1:1
+- `wide`: 5:4
+- `extra-wide`: 16:9
+
+## Richtlijnen
+
+- Gebruik dit component om plaatjes, videos en andere elementen een vaste beeldverhouding te geven, zodat ze consistent schalen.
+- Niet gebruiken met elementen met een statische grootte (hoogte en/of breedte), dan klopt de beeldverhouding niet.
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/aspect-ratio/README.md:16]: I agree, that's not very clear. I'm basically trying to say: the contents of AspectRatio should have width: 100%. Any idea on how to simply convey that?

File: storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      },
+    },
+  },
+} satisfies Meta<typeof AspectRatio>
+
+export default meta
+
+type Story = StoryObj<typeof meta>
+
+const textStyle = {
```
[the remark this answers is not in the document]
Comment (alimpens) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx:35]: I kinda like to make it very explicit what the keywords ('extra-tall', 'tall' etc.) mean in terms of aspect ratio, which is why I chose this approach. Do you think it's not necessary to make it explicit here? Or do you prefer to make it explicit in a different way?

File: storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    <AspectRatio ratio={ratio} style={{ maxWidth: getStoryConfig(ratio).maxWidth }}>
+      <img key={1} src={getStoryConfig(ratio).image} style={{ width: '100%' }} />
+      <Paragraph key={2} style={textStyle}>
+        {getStoryConfig(ratio).text}
+      </Paragraph>
+    </AspectRatio>
+  ),
+}
+
+export const Default: Story = {
```
[the remark this answers is not in the document]
Comment (alimpens) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx:98]: What 'Default' shows is dependent on the controls. If you set the controls to 'extra-wide', that's what you get. It's weird imo to show that under the heading 'Square'

File: packages/css/src/aspect-ratio/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages/css/src/aspect-ratio/README.md:16]: Let’s make it easy for ourselves and remove it for now.

```suggestion```

We’re already working on Card, and on Image in one or two weeks, and a prototype of our website in the browser. We’ll find out soon enough how to use and document this.

File: storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx:35]: I think the list in the documentation is very clear.

File: storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  backgroundColor: 'rgba(0, 0, 0, 0.6)',
+  color: 'white',
+  left: '50%',
+  padding: '16px',
+  position: 'absolute' as 'absolute',
+  top: '50%',
+  transform: 'translate(-50%, -50%)',
+}
+
+function getStoryConfig(ratio: string | undefined) {
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx]: You may have missed this suggestion?

File: storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (alimpens) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx:35]: I removed the overlay, and turned the list in the docs to a table

File: storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (alimpens) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx]: Ah, you're right. I don't quite understand the suggestion, this is a function that returns an object? Why not just define an object straight away?

```
const storyConfig = {
  "extra-tall: {},
  "tall": {},
  ...
}
```

[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.stories.tsx]: Sure!

File: storybook/storybook-react/src/AspectRatio/AspectRatio.docs.mdx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+<Meta of={AspectRatioStories} />
+
+<Markdown>{README}</Markdown>
+
+<Primary />
+
+<Controls />
+
+## Extra Tall
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/storybook-react/src/AspectRatio/AspectRatio.docs.mdx]: Sorry, can you do the keywords table as well? And the options for `ratio` in the args table?

**Review by VincentSmedinga — CHANGES_REQUESTED** [review]

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**